Fix stm32f429i disc1 doc

passgat opened the pull request:
The two patches both modify the documentation for the stm32f429i_disc1 board, fixing the flashing information and adding LTDC to the list of hardware supported in Zephyr.

passgat committed:
boards: st: stm32f429i_disc1: doc: fix flashing info

The commit e17e54c48f64 ("boards: stm32: add flashing with stm32cubeprog
for all") did not update the documentation regarding the flashing
section, which no longer uses the OpenOCD runner by default.

Fixes: e17e54c48f643dd0fdd79b754e8df6bbd662cb88
Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>

passgat committed:
boards: st: stm32f429i_disc1: doc: support LTDC hardware

The building and flashing of the drivers/display and subsys/display/lvgl
examples confirmed that the LTDC peripheral is supported.

Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>

JarmouniA reviewed (commented):
Nit: for doc consistency & clarity, could you move
```
The STM32F429I-DISC1 Discovery kit includes a ST-LINK/V2-B embedded debug tool interface.
```
to the beginning of `Programming and Debugging`,
and the flashing tools information to `Flashing` as done here https://github.com/zephyrproject-rtos/zephyr/pull/78448/commits/5276386d26d60ea77eb0e50fd2c48509974ee045 in STM32H750B-DK doc file.
Also,
```
This interface is supported by the openocd version included in Zephyr SDK.
```
could be removed.
Thanks!

passgat committed:
boards: st: stm32f429i_disc1: doc: re-work flashing section

The patch modifies the "Programming and Debugging" and "Flashing" sections
to make the documentation clearer and consistent with what is reported for
other ST boards.

Suggested-by: Abderrahmane Jarmouni <git@jarmouni.me>
Signed-off-by: Dario Binacchi <dario.binacchi@amarulasolutions.com>

passgat commented:
Gentle ping.

Thanks,
Dario
